python/ray/serve/_private/autoscaling_policy.py

```diff
@@ -202,3 +227,54 @@ def get_decision_num_replicas(
             self.decision_counter = 0
 
         return decision_num_replicas
+
+    def set_target_capacity(
+        self,
+        new_target_capacity: Optional[float],
+        new_target_capacity_direction: Optional[TargetCapacityDirection],
+    ):
+        self.target_capacity = new_target_capacity
+        self.target_capacity_direction = new_target_capacity_direction
+
+    def get_capacity_adjusted_min_replicas(self) -> int:
+        return get_capacity_adjusted_num_replicas(
+            self.config.min_replicas, self.target_capacity
+        )
+
+    def get_capacity_adjusted_initial_replicas(self) -> Optional[int]:
+        if self.config.initial_replicas is None:
+            return None
+        else:
+            return get_capacity_adjusted_num_replicas(
+                self.config.initial_replicas, self.target_capacity
+            )
+
+    def get_capacity_adjusted_max_replicas(self) -> int:
+        return get_capacity_adjusted_num_replicas(
+            self.config.max_replicas, self.target_capacity
+        )
+
+    def apply_initial_bounds(self, curr_target_num_replicas: int) -> int:
```

Comment:
it's unclear to me what "initial" means here

Reply:
This was meant to apply the bounds when a deployment was originally deployed. I've refactored the logic a bit (see [`deployment_state.py`](https://github.com/ray-project/ray/pull/41317/files#diff-c4c2583a4c2f3a3c87ada6faebd0b2dfef404e165df133a11195be5d65fcb387R1472-R1525) and [`autoscaling_policy.py`](https://github.com/ray-project/ray/pull/41317/files#diff-02afedfa9ea08814a5f985924237f709cb5eea88d611d4687f34dd617dddcc01R260-R279)), so hopefully it's easier to follow. Let me know what you think.

Reply:
Much better!